[BD-4] Convert WordCloud XModule into WordCloud XBlock. [SE-2722]

symbolist opened the pull request:
Converts the WordCloud XModule into an XBlock.

Part of [XModule to XBlock Conversion work](https://openedx.atlassian.net/wiki/spaces/AC/pages/1472790755/XModule+to+XBlock+Conversion). 

**Testing instructions**:

1. Create and edit Word Cloud XBlocks in studio.
2. Interact with them in LMS.
3. Check the behavior matches what is described in the [docs](https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/word_cloud.html).

edx-webhook commented:
Thanks for the pull request, @symbolist! I've created [OSPR-4706](https://openedx.atlassian.net/browse/OSPR-4706) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

nizarmah commented on common/lib/xmodule/xmodule/word_cloud_module.py:
```diff
@@ -91,12 +103,32 @@ class WordCloudFields(object):
+        'js': [
+            resource_string(__name__, 'assets/word_cloud/src/js/word_cloud.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js')
```
If possible, please add a trailing space for the `xmodule_js`, it seems to be forgotten since you added it elsewhere.

nizarmah commented on common/lib/xmodule/xmodule/word_cloud_module.py:
```diff
@@ -91,12 +103,32 @@ class WordCloudFields(object):
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+    studio_view_css = {
+        'scss': []
```
As noted previously, please add a trailing comma.

nizarmah commented on common/lib/xmodule/xmodule/tests/test_word_cloud.py:
```diff
@@ -1,35 +1,46 @@
-from . import LogicTest
 
+class WordCloudBlockTest(TestCase):
+    """Logic tests for Word Cloud XBlock."""
```
Since the pull request is against upstream and a slight change has already been done to that, would it be possible to convert those docstrings to multiline docstrings? 

I don't want this request to cause merge conflicts, so if you consider it would cause trouble for people who are working on this repository before your pull request is merged, then forget about it. Otherwise, please let know so I can add a comments on the other single docstrings that were changed.

symbolist replied:
According to https://www.python.org/dev/peps/pep-0008/#id33 single line docstrings are fine too.

stvstnfrd replied:
@symbolist The edx Python style guide extends PEP-8 and prescribes multi-line docstrings all the time:
- https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings

symbolist commented:
@nizarmah Thanks for your review!

@ormsbee Just warning up here. :) How are we planning to handle the review process?

stvstnfrd commented on common/lib/xmodule/xmodule/word_cloud_module.py:
```diff
@@ -37,8 +45,12 @@ def pretty_bool(value):
-    """XFields for word cloud."""
+class WordCloudBlock(  # pylint: disable=abstract-method
+    EmptyDataRawMixin, XmlMixin, EditingMixin,
+    XModuleDescriptorToXBlockMixin, XModuleToXBlockMixin, HTMLSnippet, ResourceTemplates, XModuleMixin,
```
Nit: If we put these on their own lines, like the `import`s above, it reduces the churn and subsequent chance of conflicts if/when we modify this `class`.

ormsbee commented:
@symbolist: I think you can tag myself or @stvstnfrd on these.

symbolist commented:
@ormsbee @stvstnfrd Thanks! I'll wait for the full review and then address the review comments left above.

Unfortunately, sandbox provisioning of the master branch is failing due to an issue in the configuration repo (which is being investigated by our team), so I am unable to provide our standard sandbox for testing right now.

stvstnfrd commented:
@symbolist Nothing else jumps out.

ormsbee commented:
@symbolist: Just pinging to make sure you saw that our review is done. Thank you.

symbolist commented:
@stvstnfrd @ormsbee Thanks for the review! I have addressed all feedback. 

> how OLX serialization compatibility was tested?

I have added an import-export round-trip test to cover that using an OLX version from an ironwood instance. Does that look good?

ormsbee commented:
This was rolled back due to a potential issue with codejail. It is unlikely that this PR was the cause of the regression we saw, but we'll investigate more on Monday.

symbolist commented:
@ormsbee Thanks, noted. :)
